src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -273,24 +325,29 @@ private synchronized void updateNavigatorField()
     }
 
     /**
-     * Initiates basic citizen tasks.
+     * Initiates citizen tasks
+     * Suppressing Sonar Rule Squid:S881
+     * The rule thinks we should extract ++priority in a proper statement.
+     * But in this case the rule does not apply because that would remove the readability.
      */
+    @SuppressWarnings("squid:S881")
     private void initTasks()
     {
-        this.tasks.addTask(0, new EntityAISwimming(this));
+        int priority = 0;
+        this.tasks.addTask(priority, new EntityAISwimming(this));
 
         if (this.getColonyJob() == null || !"com.minecolonies.coremod.job.Guard".equals(this.getColonyJob().getName()))
         {
-            this.tasks.addTask(1, new EntityAICitizenAvoidEntity(this, EntityMob.class, DISTANCE_OF_ENTITY_AVOID, LATER_RUN_SPEED_AVOID, INITIAL_RUN_SPEED_AVOID));
+            this.tasks.addTask(++priority, new EntityAICitizenAvoidEntity(this, EntityMob.class, DISTANCE_OF_ENTITY_AVOID, LATER_RUN_SPEED_AVOID, INITIAL_RUN_SPEED_AVOID));
         }
-        this.tasks.addTask(2, new EntityAIGoHome(this));
-        this.tasks.addTask(3, new EntityAISleep(this));
-        this.tasks.addTask(4, new EntityAIOpenDoor(this, true));
-        this.tasks.addTask(4, new EntityAIOpenFenceGate(this, true));
-        this.tasks.addTask(5, new EntityAIWatchClosest2(this, EntityPlayer.class, 3.0F, 1.0F));
-        this.tasks.addTask(6, new EntityAIWatchClosest2(this, EntityCitizen.class, 5.0F, 0.02F));
-        this.tasks.addTask(7, new EntityAICitizenWander(this, 0.6D));
-        this.tasks.addTask(8, new EntityAIWatchClosest(this, EntityLiving.class, 6.0F));
+        this.tasks.addTask(++priority, new EntityAIGoHome(this));
+        this.tasks.addTask(++priority, new EntityAISleep(this));
+        this.tasks.addTask(++priority, new EntityAIOpenDoor(this, true));
+        this.tasks.addTask(priority, new EntityAIOpenFenceGate(this, true));
+        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityPlayer.class, 3.0F, 1.0F));
+        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityCitizen.class, 5.0F, 0.02F));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Assign this magic number 5.0F to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Assign this magic number 0.02F to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)